.github/workflows/publish_prod.yml

```diff
@@ -0,0 +1,36 @@
+name: publish_prod
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+  # In case an automated release failed, we can retrigger them manually.
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  publish:
+    # To avoid releasing a wrong version: check this commit is tagged
+    if: startsWith(github.event.workflow_run.head_branch, 'refs/tags/v') }}
```

Comment:
use main workflow instead of a differnt workflow, this allows us to reuse artifacts more easily, + rebuild more easily:
```
push:
    tags:
      - "v*.*.*"
```
+ 
```
if: startsWith(github.event.workflow_run.head_branch, 'refs/tags/v') }}
```